extensions/api/data-management/contractdefinition/src/test/java/org/eclipse/dataspaceconnector/api/datamanagement/contractdefinition/service/ContractDefinitionEventDispatchTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.api.datamanagement.contractdefinition.service;
+
+import org.eclipse.dataspaceconnector.junit.extensions.EdcExtension;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.event.Event;
+import org.eclipse.dataspaceconnector.spi.event.EventRouter;
+import org.eclipse.dataspaceconnector.spi.event.EventSubscriber;
+import org.eclipse.dataspaceconnector.spi.event.contractdefinition.ContractDefinitionCreated;
+import org.eclipse.dataspaceconnector.spi.event.contractdefinition.ContractDefinitionDeleted;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(EdcExtension.class)
+public class ContractDefinitionEventDispatchTest {
+
+    private final EventSubscriber eventSubscriber = mock(EventSubscriber.class);
+
+    @Test
+    void shouldDispatchEventOnContractDefinitionCreationAndDeletion(ContractDefinitionService service, EventRouter eventRouter) throws InterruptedException {
+        var createdLatch = onDispatchLatch(ContractDefinitionCreated.class);
+        var deletedLatch = onDispatchLatch(ContractDefinitionDeleted.class);
+        eventRouter.register(eventSubscriber);
+        var contractDefinition = ContractDefinition.Builder.newInstance()
+                .id(UUID.randomUUID().toString())
+                .contractPolicyId(UUID.randomUUID().toString())
+                .accessPolicyId(UUID.randomUUID().toString())
+                .selectorExpression(AssetSelectorExpression.SELECT_ALL)
+                .build();
+
+        service.create(contractDefinition);
+
+        assertThat(createdLatch.await(10, SECONDS)).isTrue();
```

Comment:
Since we have `awaitility` already in the codebase, I think we should use it in async tests

Reply:
@paullatzelsperger agree, I will open an issue about replacing all the latches with awaitility, just to have an unique way to test async code.

Reply:
thanks, good for me.